hack/tools.mk

```diff
@@ -26,5 +27,5 @@ export PATH := $(abspath $(TOOLS_BIN_DIR)):$(PATH)
 # Tools                                 #
 #########################################
 
-$(GINKGO): go.mod
```

Comment:
There are make targets like `test`, `test-cov`, `test-integration` which currently depends upon `GINKGO` dependency. What is the relation between removing Ginkgo and adding Kustomize targets in tools.mk?

Reply:
@unmarshall There are two things this PR fixes,
- Installs the kustomize dependency for Etcd-druid
- One more thing fixed as part of this PR is to fix a make warning, by removing the `GINKGO` target from etcd-druid/hack/tools.mk.
This is because the `GINKGO` target is always overriding by `vendor/github.com/gardener/gardener/hack/tools.mk:110`. So, we don't need this `GINKGO` target from `etcd-druid/hack/tools.mk`.

Reply:
Sorry for confusion, I updated the PR description

Reply:
If we plan to eventually remove druid's dependency on gardener/gardener hack scripts, as described by https://github.com/gardener/etcd-druid/issues/475, then we should ideally be retaining the GINKGO target in druid and instead work on removing the dependency on g/g GINKGO target, to avoid this conflict warning.

Reply:
I agree. @shreyas-s-rao would it make sense to get this change in. Removing dependency on g/g scripts is taking a bit long. I will introduce it again later on once the dependency has been removed.

Reply:
Sure, we can remove druid's GINKGO target in this PR, as we anyway plan to move necessary targets from g/g to druid later on.